protostar/cheatable_starknet/callable_hint_locals/declare_hint_local.py

```diff
@@ -0,0 +1,71 @@
+import asyncio
+from typing import Callable
+
+from starkware.starknet.core.os.contract_class.compiled_class_hash import (
+    compute_compiled_class_hash,
+)
+
+from protostar.cairo.short_string import short_string_to_str
+from protostar.cheatable_starknet.callable_hint_locals.callable_hint_local import (
+    CallableHintLocal,
+)
+from protostar.cheatable_starknet.controllers.contracts import (
+    DeclaredContract,
+    ContractsController,
+    DeclaredSierraClass,
+)
+from protostar.compiler import CompilationException
+from protostar.compiler.project_compiler import ProjectCompiler
+from protostar.configuration_file.configuration_file import (
+    ContractNameNotFoundException,
+)
+from protostar.starknet import CheatcodeException
+
+
+class DeclareHintLocal(CallableHintLocal):
+    def __init__(
+        self,
+        contracts_controller: ContractsController,
+        project_compiler: ProjectCompiler,
+    ):
+        self._contracts_controller = contracts_controller
+        self._project_compiler = project_compiler
+
+    @property
+    def name(self) -> str:
+        return "declare"
+
+    def _build(self) -> Callable:
+        return self.declare
+
+    def declare(self, contract: int) -> DeclaredContract:
```

Comment:
There shouldn't be this much code in `HintLocal`, can it be moved to the controller? (refer to `DeployHintLocal` as per how much code should be in the hintlocal itself)